st2common/st2common/services/sensor_watcher.py

```diff
@@ -95,9 +95,18 @@ def start(self):
             self.connection.release()
 
     def stop(self):
+        LOG.debug('Shutting down sensor watcher.')
         try:
             if self._updates_thread:
                 self._updates_thread = eventlet.kill(self._updates_thread)
+
+            if self.connection:
+                channel = self.connection.channel()
```

Comment:
>in this case queue will be deleted when connection to the broken is lost.

This is exactly the problem I have. They detect server restarts with connection drops which is entirely incorrect. I have seen connection drops in AWS so many times and we have code to reestablish connection. Deleting the queue could mean we lose information about some sensors. In the explicit delete case, a queue is deleted exactly when the sensor container is stopped. And on restart, we anyways load all the sensors and therefore we won't miss anything. The only downside is stray queues maybe left around if the process crashes the chance of which is fairly low.